src/data/error.rs

```diff
@@ -287,6 +324,45 @@ pub enum LexError {
     #[error("no newline at end of file")]
     NoNewlineAtEOF,
 
+    #[error("unknown token: {0:?}")]
+    UnknownToken(u8),
+
+    #[error("missing terminating \" character in string literal")]
+    StringMissingEndQuote,
+
+    #[error("illegal newline while parsing string literal")]
+    NewlineInString,
+
+    #[error("{0} character escape out of range")]
+    InvalidNumericCharEscape(Radix),
+
+    #[error("overflow while parsing integer literal")]
+    IntegerOverflow,
+
+    #[error("exponent for floating literal has no digits")]
+    FloatExponentMissingDigits,
+
+    #[error("missing digits to {0} integer constant")]
+    MissingDigits(Radix),
+
+    #[error("{0}")]
+    ParseFloat(#[from] std::num::ParseFloatError),
```

Comment:
Oh this is clever! I wouldn't have thought of this :)

Reply:
I plan to do this with hexponent's error type too.